jekyll/_cci2/high-uid-error.md

```diff
@@ -8,54 +8,70 @@ order: 21
-
-When you hit this error, you need to update the files' UID/GID and re-create the image.
-
-Please contact the image maintainer and report the error details.
-
-If you are an image maintainer, you need to look for which file has high UID/GID and correct it accordingly.
-
-Here is one way to find such a file inside [circleci/doc-highid](https://hub.docker.com/r/circleci/doc-highid).
-
-First, grab the high UID/GID value from `container id XXX cannot be mapped to a host id` error message in CircleCI.
-For example, the value is `1000000` in the error message you see at the top of this document.
-
-Second, start the container and look for which files get the high value. There are multiple ways to do this, but one way is using `find` command. Note that the following example only works with BSD/GNU `find` command. If BSD/GNU `find` is not installed in your container, you may need to use different commands.
-
-```
+This document explains the problem and how to fix it.
+
+## Background
+
+The user namespace (`userns`) is a feature of the Linux kernel
+that adds another security layer to Linux containers.
+The `userns` allows a host machine
+to run containers outside its UID/GID namespace.
+This means all containers can have a root account (UID 0) in their own namespace
+and run processes without receiving root privileges from the host machine.
+
+When a `userns` is created,
+the Linux kernel provides a mapping between the container and the host machine.
+For example,
+if you start a container
+and run a process with UID 0 inside of it,
+the Linux kernel maps the container's UID 0 to a non-privileged UID on the host machine.
+This allows the container to run a process as if it were the root user,
+while **actually** being run by the non-root user on the host machine.
+
+## Problem
+
+The error is caused by a `userns` remapping failure.
+CircleCI runs Docker containers with `userns` enabled
+in order to securely run customers' containers.
+The host machine is configured with a range of valid UID/GID for remapping.
+
+When Docker starts a container,
+Docker pulls an image
+and extracts layers from that image.
+If a layer contains files with UID/GID outside the defined range,
+Docker cannot successfully remap
+and fails to start the container.
+
+## Solution
+
+To fix this error,
+you must update the files' UID/GID
+and re-create the image.
+
+If you are not the image maintainer, congratulations:
+it's not your responsibility.
+Contact the imagine maintainer
+and report the error.
+
+If you are the image maintainer,
+identify the file with the high UID/GID
+and correct it.
+First, grab the high UID/GID from the error message.
+In the example presented at the top of this document,
+this value is `1000000`.
+Next, start the container and look for the files which receive that invalid value.
+
+One way to do this is to use the `find` command,
+as shown below:
+
+```bash
 # Start a shell inside the container
 $ docker run -it circleci/doc-highid sh
```

Comment:
Maybe we should address https://github.com/circleci/circleci-docs/issues/1639 as well in this PR?